File: src/concurrency/book2/sample/chapter8/java/DefaultThreadFactory.java
package concurrency.book2.sample.chapter8.java;

import java.util.concurrent.atomic.AtomicInteger;

public class DefaultThreadFactory implements ThreadFactory {

    private static final AtomicInteger GROUP_COUNTER = new AtomicInteger();

    private static final ThreadGroup group = new ThreadGroup("MyThreadPool-" + GROUP_COUNTER.getAndIncrement());

    private static final AtomicInteger COUNTER = new AtomicInteger(0);

    @Override
    public Thread createThread(Runnable runnable) {
        return new Thread(group,runnable,"thread-pool-" + COUNTER.getAndIncrement());
    }
}

File: src/concurrency/book2/sample/chapter8/kotlin/DefaultThreadFactory.kt
package concurrency.book2.sample.chapter8.kotlin

import java.util.concurrent.atomic.AtomicInteger

class DefaultThreadFactory : ThreadFactory {

    private companion object {
        val GROUP_COUNTER: AtomicInteger = AtomicInteger()
        val group:ThreadGroup = ThreadGroup("MyThreadPool-" + GROUP_COUNTER.getAndIncrement())
        val COUNTER = AtomicInteger(0)
    }

    override fun createThread(runnable: Runnable): Thread {
        return Thread(group,runnable,"thread-pool-" + COUNTER.getAndIncrement())
    }
}